src/core/valuemapmodelbase.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+                            valuemapmodelbase.h
+
+                              -------------------
+              begin                : March 2019
+              copyright            : (C) 2019 by Matthias Kuhn
+              email                : matthias@opengis.ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef VALUEMAPMODELBASE_H
+#define VALUEMAPMODELBASE_H
+
+#include <QAbstractListModel>
+#include <QVariant>
+
+/**
+ * A model that manages the key/value pairs for a ValueMap widget.
+ */
+class ValueMapModelBase : public QAbstractListModel
+{
+    Q_OBJECT
+
+    /**
+     * A list of QVariantMap, wrapped in a QVariant.
+     *
+     * Like this:
+     *
+     * [{'CH': 'Switzerland'}, {'DE': 'Germany'}, {'FR': 'France'}]
+     */
+    Q_PROPERTY( QVariant valueMap READ map WRITE setMap NOTIFY mapChanged )
```

Comment:
you can remove that, it's not going to be exposed in QML anyways.

Reply:
Yes it is used here https://github.com/opengisch/QField/blob/3b03fff3432ed7b7dc618358f627dca404f00309/src/qml/editorwidgets/ValueMap.qml#L49C7-L49C12

Reply:
Oh I see now. In `AttributeFormModelBase` there where also some unused property. I removed them as well.